flink-runtime/src/main/java/org/apache/flink/runtime/minicluster/MiniCluster.java

```diff
@@ -1361,8 +1364,15 @@ public void onFatalError(Throwable exception) {
 
     private class ShutDownFatalErrorHandler implements FatalErrorHandler {
 
+        private final Consumer<Throwable> customFatalErrorHandler;
+
+        public ShutDownFatalErrorHandler(Consumer<Throwable> customFatalErrorHandler) {
+            this.customFatalErrorHandler = customFatalErrorHandler;
+        }
+
         @Override
         public void onFatalError(Throwable exception) {
+            customFatalErrorHandler.accept(exception);
```

Comment:
since isn't really an error handler, for the simple reason that the ShutDownFatalErrorHandler is still called and will still kill the mini cluster. rephrase it to "fatalErrorListener".